### PR TITLE
Implement `From<Arc<LogicalPlan>>` for `LogicalPlanBuilder`

### PR DESCRIPTION
**Pull request by AbrarNitk**
## Which issue does this PR close?

Closes  https://github.com/apache/datafusion/issues/10465

## What changes are included in this PR?

This PR implements `From` trait for converting `Arc<LogicalPlan>` into `LogicalPlanBuilder`

## Are these changes tested?

Yes

## Are there any user-facing changes?

No

### CODE REVIEW

**Comment by alamb** [comment]
Thanks @AbrarNitk ! I started the CI checks on this PR 

Normally I think we should add some test coverage so we don't accidentally break this in the future. 

This would I think this could take the form of:
1. (Ideally) find an example of wat
1. Add a doc example
2. Add a unit test

However, in this case perhaps I think we could simply merge this PR and then use the code in https://github.com/apache/datafusion/pull/10460 with @ClSlaid

Looks like the CI failed for running `rust fmt` https://github.com/apache/datafusion/actions/runs/9050685356/job/24892712284?pr=10466

Can you fix that and then we can get this PR merged?

Thanks agian

**Comment by AbrarNitk** [comment]
Hey @alamb, I am glad you have checked PR. I am new to arrow and `datafusion` and `arrow`, will keep learning and also keep sending the PRs :). Thank you 🙏

**Comment by alamb** [comment]
> Hey @alamb, I am glad you have checked PR. I am new to `datafusion` and `arrow`, will keep learning and also keep sending the PRs :). Thank you 🙏

Thank you and welcome to the community! I started CI again

File: datafusion/expr/src/logical_plan/builder.rs [hunk 1]
```diff
@@ -1138,6 +1139,31 @@ impl LogicalPlanBuilder {
         )?))
     }
 }
+
+/// Converts a `Arc<LogicalPlan>` into `LogicalPlanBuilder`
```
Comment (alamb) [inline datafusion/expr/src/logical_plan/builder.rs:1143]: FYI in general, the style of docs in Rust is to put them in triple quotes which will ensure they are actually run as part of the CI tests (`cargo test --doc` will run them)

So in this case like 

```
/// ```
/// fn employee_schema() -> Schema {
...
```

Any chance you are willing to make a follow on PR to change this to a docstring @AbrarNitk ?

**Review by alamb — APPROVED** [review]
Thanks @AbrarNitk  -- I think this is a nice improvement and a good first contribution ❤️ 

Merging it in so we can potentially use it in  https://github.com/apache/datafusion/pull/10460

File: datafusion/expr/src/logical_plan/builder.rs (hunk 1 above)
Comment (AbrarNitk) [reply datafusion/expr/src/logical_plan/builder.rs:1143]: Yep, I will raise another PR.

Comment (AbrarNitk) [reply datafusion/expr/src/logical_plan/builder.rs:1143]: Hey @alamb, please check the PR: https://github.com/apache/datafusion/pull/10466

**Pull request merged**